backend/src/Manager/Subscription/SubscriptionHistoryManager.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace App\Manager\Subscription;
+
+use App\AutoMapping;
+use App\Entity\SubscriptionHistoryEntity;
+use App\Repository\SubscriptionHistoryEntityRepository;
+use App\Request\Subscription\SubscriptionHistoryCreateRequest;
+use Doctrine\ORM\EntityManagerInterface;
+use DateTime;
+
+class SubscriptionHistoryManager
+{
+    public function __construct(private AutoMapping $autoMapping, private EntityManagerInterface $entityManager, private SubscriptionHistoryEntityRepository $subscribeHistoryRepository)
+    {
+        $this->autoMapping = $autoMapping;
+        $this->entityManager = $entityManager;
+        $this->subscribeHistoryRepository = $subscribeHistoryRepository;
+    }
+
+    public function createSubscriptionHistory($subscription):SubscriptionHistoryEntity
+    { 
+        $request = new SubscriptionHistoryCreateRequest();
```

Comment:
Why not injecting the request object of _SubscriptionHistoryCreateRequest_ in the function's header instead of instantiating a new one?

Reply:
This function requires this, done added define the type of the var SubscriptionEntity